client/finality-grandpa/src/aux_schema.rs

```diff
@@ -500,26 +504,29 @@ where
 /// We always keep around the justification for the best finalized block and overwrite it
 /// as we finalize new blocks, this makes sure that we don't store useless justifications
 /// but can always prove finality of the latest block.
-pub(crate) fn update_best_justification<Block: BlockT, F, R>(
-	justification: &GrandpaJustification<Block>,
+pub(crate) fn update_best_justification<Header, J, F, R>(
+	justification: &J,
 	write_aux: F,
 ) -> R
 where
+	Header: HeaderT,
+	J: ProvableJustification<Header>,
```

Comment:
I wanted to remove `ProvableJustification` as it is a bit useless right now. Tests can just create a proper `GrandpaJustification`.